### PR TITLE
Feature | Notifications mechanism

### PR DESCRIPTION
**Pull request by AlanSoto31**
## Description
- Implement a reminder system to notify users at specific intervals before an event begins. Initially, the reminders will be set for 15 minutes, 10 minutes, and at the event's start time. 
- [Noticed gem](https://github.com/excid3/noticed) setup.

- [Task](https://github.com/orgs/TelosLabs/projects/5/views/1?pane=issue&itemId=68736380)
## How has this been tested?

Please mark the tests that you ran to verify your changes. If difficult to test, consider providing instructions so reviewers can test.

- [x] Manual testing
- [ ] System tests
- [ ] Unit tests
- [ ] None

## Checklist

- [x] CI pipeline is passing
- [x] My code follows the conventions of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added seed data to the database (if applicable)

## How to test

1. Update the REMINDER_CANDENCE constant as below:
```
  REMINDER_CADENCE = {
    first_reminder: 2.minutes,
    second_reminder: 1.minute,
    last_reminder: 0.minutes
  }.
```
2. Ensure `User.last.profile.mail_notifications = true`
3. Run the code below to create some events for testing :
```
(1..6).each_with_index do |cadence, index|
  starts_at = Time.zone.now + cadence.minutes

  event = Event.create!(
    conference: Conference.last,
    title: "Event #{index + 1}",
    description: Faker::Lorem.paragraph,
    starts_at: starts_at,
    ends_at: (starts_at + 10.minutes),
    location: Location.last
  )

  event.users << User.last
end
```
4. The email notifications for each event should be delivered like the table below

image

### CODE REVIEW

File: app/models/event/reminder_scheduler.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module Event::ReminderScheduler
+  extend ActiveSupport::Concern
+
+  REMINDER_CADENCE = {
+    first_reminder: 15.minutes,
+    second_reminder: 10.minutes,
+    last_reminder: 0.minutes
+  }.freeze
```
Comment (AlanSoto31) [inline app/models/event/reminder_scheduler.rb:8]: Should the reminder cadence be editable for admins?

File: app/notifiers/event_notifier.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class EventNotifier < ApplicationNotifier
+  deliver_by :email do |config|
+    config.mailer = "EventMailer"
+    config.method = "reminder"
+    config.if = -> { recipient.profile.mail_notifications }
+  end
```
Comment (AlanSoto31) [inline app/notifiers/event_notifier.rb:6]: Push notifications will be set in a separate PR.

File: app/notifiers/event_notifier.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class EventNotifier < ApplicationNotifier
+  deliver_by :email do |config|
+    config.mailer = "EventMailer"
+    config.method = "reminder"
+    config.if = -> { recipient.profile.mail_notifications }
```
Comment (Sergio-e) [inline app/notifiers/event_notifier.rb:5]:
```suggestion
    config.if = -> { recipient.profile&.mail_notifications }
```

File: app/models/event/reminder_scheduler.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  included do
+    after_save_commit :schedule_reminder
+  end
+
+  private
+
+  def schedule_reminder
```
Comment (Sergio-e) [inline app/models/event/reminder_scheduler.rb:16]: I suggest changing the approach. It's going to be cumbersome to update/remove jobs if the cadence or the event's schedule changes. You can use SolidQueue's [recurring tasks mechanism](https://github.com/rails/solid_queue/?tab=readme-ov-file#recurring-tasks) to enqueue a job that will generate the appropriate notifications.

**Review by Sergio-e — APPROVED** [review]
